operator/v1/types_machineconfiguration.go

```diff
@@ -153,3 +167,181 @@ const (
+}
+
+// NodeDisruptionPolicyConfig is the overall spec definition for files/units/sshkeys
+type NodeDisruptionPolicyConfig struct {
+	// files is a list of MachineConfig file definitions and actions to take to changes on those paths
+	// +optional
+	// +listType=map
+	// +listMapKey=path
+	Files []NodeDisruptionPolicyFile `json:"files,omitempty"`
+	// units is a list MachineConfig unit definitions and actions to take on changes to those services
+	// +optional
+	// +listType=map
+	// +listMapKey=name
+	Units []NodeDisruptionPolicyUnit `json:"units,omitempty"`
+	// sshkey is the overall sshkey MachineConfig definition
+	// +optional
+	SSHKey NodeDisruptionPolicySSHKey `json:"sshkey,omitempty"`
+}
+
+// NodeDisruptionPolicyClusterStatus is the type for the status object, rendered by the controller as a
+// merge of cluster defaults and user provided policies
+type NodeDisruptionPolicyClusterStatus struct {
+	// files is a list of MachineConfig file definitions and actions to take to changes on those paths
+	// +optional
+	// +listType=map
+	// +listMapKey=path
+	Files []NodeDisruptionPolicyFile `json:"files,omitempty"`
+	// units is a list MachineConfig unit definitions and actions to take on changes to those services
+	// +optional
+	// +listType=map
+	// +listMapKey=name
+	Units []NodeDisruptionPolicyUnit `json:"units,omitempty"`
+	// sshkey is the overall sshkey MachineConfig definition
+	// +optional
+	SSHKey NodeDisruptionPolicySSHKey `json:"sshkey,omitempty"`
+}
+
+// NodeDisruptionPolicyFile is a file entry and corresponding actions to take
+type NodeDisruptionPolicyFile struct {
+	// path represents the filepath
+	// +kubebuilder:validation:Required
+	Path string `json:"path"`
+	// actions represents the series of commands to be executed on changes to the corresponding type and value
+	// +kubebuilder:validation:Required
+	// +listType=atomic
+	Actions []NodeDisruptionPolicyAction `json:"actions"`
+}
+
+// NodeDisruptionPolicyUnit is a systemd unit name and corresponding actions to take
+type NodeDisruptionPolicyUnit struct {
+	// name represents the service name
+	// +kubebuilder:validation:Required
+	Name string `json:"name"`
+	// actions represents the series of commands to be executed on changes to the corresponding type and value
+	// +kubebuilder:validation:Required
+	// +listType=atomic
+	Actions []NodeDisruptionPolicyAction `json:"actions"`
+}
+
+// NodeDisruptionPolicySSHKey is actions to take for any SSHKey change
+type NodeDisruptionPolicySSHKey struct {
+	// actions represents the series of commands to be executed on changes to the corresponding type and value
+	// +kubebuilder:validation:Required
+	// +listType=atomic
+	Actions []NodeDisruptionPolicyAction `json:"actions"`
+}
+
+// Validation is a string enum used in a NodeDisruptionPolicyStatus object. They describe whether the controller
+// was able to validate the latest spec
+// +kubebuilder:validation:Enum:="success";"failed"
```

Comment:
Enums like this should be PascalCase, use `Success` and `Failed`

Reply:
Removed in favour of having a more proper condition, instead of a "success/fail" flag